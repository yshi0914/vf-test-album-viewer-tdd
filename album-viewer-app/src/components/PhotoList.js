import React from "react";
import ThumbnailPhoto from "./ThumbnailPhoto";

const PhotoList = (props) => {
  const { data } = props;
  return (
    <div className="row" id="photolist">
      {data &&
        data.map((d) => {
          return <ThumbnailPhoto key={d.id} data={d} />;
        })}
    </div>
  );
};

export default PhotoList;
