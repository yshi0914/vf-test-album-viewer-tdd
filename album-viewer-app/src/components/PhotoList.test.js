import React from "react";
import PhotoList from "./PhotoList";
import { shallow, mount } from "enzyme";
import photosData from "./photosTest.json";
import { BrowserRouter as Router } from "react-router-dom";
// All the photos from the target album should be put under one <div> tag
describe("The number of album list to be rendered should be one", () => {
  let wrapper = mount(
    <Router>
      <PhotoList data={photosData} />
    </Router>
  );
  test("render the big title of this Album Viewer App", () => {
    expect(wrapper.find("#photolist")).toHaveLength(1);
  });
});

// since there are 50 photos in the given photos file, the number of <ThumbnailPhoto> tag should be 50
describe("The number of album list to be rendered should be one", () => {
  let wrapper = mount(
    <Router>
      <PhotoList data={photosData} />
    </Router>
  );
  test("check the number of photos loaded into the page", () => {
    expect(wrapper.find("ThumbnailPhoto")).toHaveLength(50);
  });
});

// Title name of one of those testing photos: 'natus doloribus necessitatibus ipsa'
describe("the title name should be rendered correctly", () => {
  let wrapper = mount(
    <Router>
      <PhotoList data={photosData} />
    </Router>
  );
  test("check whether the title name is correctly displayed", () => {
    expect(wrapper.contains("natus doloribus necessitatibus ipsa")).toEqual(
      true
    );
  });
});

// Snapshot Testing for PhotoList component
describe("PhotoList component snapshot testing", () => {
  test("matches the snapshot", () => {
    const component = shallow(
      <Router>
        <PhotoList data={photosData[0]} />
      </Router>
    );
    expect(component.getElements()).toMatchSnapshot();
  });
});
