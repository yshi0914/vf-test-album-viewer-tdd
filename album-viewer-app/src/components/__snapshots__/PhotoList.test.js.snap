// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoList component snapshot testing matches the snapshot 1`] = `
Array [
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <PhotoList
      data={
        Object {
          "albumId": 1,
          "id": 1,
          "thumbnailUrl": "https://via.placeholder.com/150/92c952",
          "title": "accusamus beatae ad facilis cum similique qui sunt",
          "url": "https://via.placeholder.com/600/92c952",
        }
      }
    />
  </Router>,
]
`;
